src/main/scala/org/broadinstitute/hail/driver/FilterVariantsExpr.scala

```diff
@@ -40,10 +38,13 @@ object FilterVariantsExpr extends Command {
     if (!(options.keep ^ options.remove))
       fatal("either `--keep' or `--remove' required, but not both")
 
-    val vas = vds.vaSignature
     val cond = options.condition
     val keep = options.keep
 
+    state.copy(vds = filterVariantsExpr(vds, cond, keep))
+  }
+
+  def filterVariantsExpr(vds: VariantDataset, cond: String, keep: Boolean): VariantDataset = {
```

Comment:
I think this might fit better as a method on VSM:
```scala
state.copy(vds = vds.filterVariantsExpr(cond, keep))
```

Should be easy to move it there.